module/tracer.go

```diff
@@ -10,14 +10,41 @@ import (
 	"github.com/onflow/flow-go/module/trace"
 )
 
+type TraceSpan opentracing.Span
+
 var _ Tracer = &trace.OpenTracer{}
+var _ Tracer = &trace.NoopTracer{}
 
 // Tracer interface for tracers in flow. Uses open tracing span definitions
 type Tracer interface {
 	ReadyDoneAware
-	StartSpan(entity flow.Identifier, spanName trace.SpanName, opts ...opentracing.StartSpanOption) opentracing.Span
-	FinishSpan(entity flow.Identifier, spanName trace.SpanName)
-	GetSpan(entity flow.Identifier, spanName trace.SpanName) (opentracing.Span, bool)
+
+	// StartBlockSpan starts an span for a block, built as a child of rootSpan
```

Comment:
When creating a span for a block, it will be sampled by the block ID. In other words, if a block is sampled, all spans for a certain block will be sampled.

This is great. However, I'm concerned about two issues:

1. If the network is big, i.e. 400 nodes, then even with sampling, then we will still aggregate spans from all of the nodes to the same span. Would it be hard to view all spans from all nodes on the chart? Can we group spans by block ID first, then node type, and then node ID?
Also with so many nodes having the same node type, I wonder if possible we can sample by node type as well. For instance, if there are 100 verification nodes, we could only pick 3 of them to report their spans. Because most likely, similar node will have similar tracing results.
The consensus nodes might be different, because the leader and next leader might have different tracing data.
So I'm not sure how easy to include the leader, next leader, and a sample of followers. :D 

2. All transaction spans are grouped under the same block, since we will be most likely looking at blocks having lots of transactions to dig into performance issue, then would it easy to open such block on the chart?

Reply:
yeah we could filter by service name, the idea is we only keep a sample of node running tracing for now. we could change the logic in the future. 

Transaction Spans are not grouped under same block, they form their own group of spans under a root span with traceId of transactionID.